### PR TITLE
Remove rowData and rowID props, use data and key

### PR DESCRIPTION
**Pull request by hanastasov**
Closes https://github.com/IgniteUI/igniteui-angular/issues/10456

### Additional information (check all that apply):
 - [ ] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [x] This PR contains breaking changes
 - [x] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Review by damyanpetev — CHANGES_REQUESTED** [review]
I see no changes to packages in the package.json so no reason to update the lock file

File: projects/igniteui-angular/src/lib/grids/grid/grid-row.component.html [hunk 1]
```diff
@@ -15,7 +15,7 @@
         <div class="igx-grid__cbx-selection igx-grid__tr-action" (pointerdown)="$event.preventDefault()" (click)="onRowSelectorClick($event)">
             <ng-template *ngTemplateOutlet="
                 this.grid.rowSelectorTemplate ? this.grid.rowSelectorTemplate : rowSelectorBaseTemplate;
-                context: { $implicit: { index: viewIndex, rowID: rowID, selected: selected }}">
+                context: { $implicit: { index: viewIndex, rowID: key, selected: selected }}">
```
[the remark this answers is not in the document]
Comment (damyanpetev) [reply projects/igniteui-angular/src/lib/grids/grid/grid-row.component.html:18]: This is in the "nice to have" category btw

File: projects/igniteui-angular/src/lib/grids/api.service.ts [hunk 2]
```diff
@@ -83,16 +83,31 @@ export class GridBaseAPIService<T extends GridType> implements GridServiceType {
...
         return this.grid.rowList.find((row) => row.index === rowIndex);
     }
 
+    /**
+     * Gets the rowID of the record at the specified data view index
+     *
+     * @param index
+     * @param dataCollection
+     */
+    public get_rec_id_by_index(index: number, dataCollection?: any[]): any {
```
[the remark this answers is not in the document]
Comment (hanastasov) [reply projects/igniteui-angular/src/lib/grids/api.service.ts:102]: Well, actually there was an error runtime:
image

File: projects/igniteui-angular/src/lib/grids/grid/grid-row.component.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (hanastasov) [reply projects/igniteui-angular/src/lib/grids/grid/grid-row.component.html]: Just mentioned in the backlog that `rowID` in the row context object is deprecated and added `key` as well. Dont know if we can really deprecate it as any other prop.

File: src/app/grid-finjs/grid-finjs.component.html [hunk 3]
```diff
@@ -52,8 +52,8 @@
         <ng-template igxCell let-cell="cell">
             <div class="finjs-icons">
                 <span>{{cell.value | currency:'USD':'symbol':'1.4-4'}}</span>
-                <igx-icon *ngIf="trends.positive(cell.row.rowData)">trending_up</igx-icon>
-                <igx-icon *ngIf="trends.negative(cell.row.rowData)">trending_down</igx-icon>
+                <igx-icon *ngIf="trends.positive(cell.row.data)">trending_up</igx-icon>
+                <igx-icon *ngIf="trends.negative(cell.row.data)">trending_down</igx-icon>
```
[the remark this answers is not in the document]
Comment (hanastasov) [reply src/app/grid-finjs/grid-finjs.component.html:56]: Will definitely add an entry about that!

File: CHANGELOG.md [hunk 4]
```diff
@@ -80,8 +80,23 @@ All notable changes for each version of this project will be documented in this
         Use `IgxGridToolbarComponent`, `IgxGridToolbarHidingComponent`, `IgxGridToolbarPinningComponent` instead.
     - **Breaking Change** - The `rowSelected` event is renamed to `rowSelectionChanging` to better reflect its function.
     - **Breaking Change** - The `columnSelected` event is renamed to `columnSelectionChanging` to better reflect its function.
-- `igxGrid`
+    - **Breaking Change** - `columnsCollection` is removed. Use `columns` instead.
     - Exposed a `groupStrategy` input that functions similarly to `sortStrategy`, allowing customization of the grouping behavior of the grid. Please, refer to the [Group By ](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/groupby) topic for more information.
+- `RowType`, `IgxRowDirective`
+    - **Breaking Change** - `rowData` and `rowID` deprecated properties are now removed. Use `data` and `key` instead. Use `ng update` for automatic migration.
+- `RowSelectorTemplate`
```
Comment (damyanpetev) [inline CHANGELOG.md:87]: Kind of hard to tell from the diff, but these bullets don't seem to be under the grids as nested level, yet they probably should be, right?

Also either "`igxRowSelector` template" or "Row Selector template" - pick one, cuz the code highlighted `RowSelectorTemplate` is not a class/selector/anything we have.
..Though looking at the other sections I don't blame you, it's a bit of a mess

File: CHANGELOG.md [hunk 5]
```diff
@@ -80,11 +80,23 @@ All notable changes for each version of this project will be documented in this
...
+    <igx-grid [data]="data", [rowSelection]="'multiple'" primaryKey="ID">
+        <igx-column field="Name"></igx-column>
+        <igx-column field="Age"></igx-column>
+
+        <ng-template igxRowSelector let-rowContext>
+            <span>{{ rowContext.key }}</span>
+        </ng-template>
+    </igx-grid>
+    ```
+`IgxColumnActionsComponent`
```
Comment (damyanpetev) [inline CHANGELOG.md:99]: duplicate `IgxColumnActionsComponent`

**Review by damyanpetev — APPROVED** [review]

**Pull request merged**